src/Native/Unix/System.Security.Cryptography.Native.Apple/pal_x509.cpp

```diff
@@ -0,0 +1,478 @@
+    }
+    else if (osStatus != errSecUnsupportedFormat)
+    {
+        printf("Call 3: %d\n", osStatus);
+    }
+
+    return PAL_X509Unknown;
+}
+
+static void ForceLoadPrivateKey(SecIdentityRef identity)
+{
+    // Exporting the key will cause it to be loaded out of the keychain before
+    // we delete the keychain.  And since the same object is kept alive across
+    // multiple calls to SecIdentityCopyPrivateKey, this makes it usable to the
+    // rest of the process
+    SecKeyRef privateKey = nullptr;
+    OSStatus local = SecIdentityCopyPrivateKey(identity, &privateKey);
+
+    if (local == noErr)
+    {
+        SecExternalFormat dataFormat = kSecFormatWrappedPKCS8;
+        SecItemImportExportKeyParameters keyParams = {};
+        keyParams.version = SEC_KEY_IMPORT_EXPORT_PARAMS_VERSION;
+        keyParams.passphrase = CFSTR("ExportPassphrase");
+
+        CFDataRef exported = nullptr;
+
+        SecItemExport(privateKey, dataFormat, 0, &keyParams, &exported);
+
+        CFRelease(keyParams.passphrase);
+
+        if (exported != nullptr)
+        {
+            CFRelease(exported);
+        }
+    }
+
+    if (privateKey != nullptr)
+    {
+        CFRelease(privateKey);
+    }
+}
+
+static int32_t ProcessCertificateTypeReturn(CFArrayRef items, SecCertificateRef* pCertOut, SecIdentityRef* pIdentityOut)
+{
+    assert(pCertOut != nullptr && *pCertOut == nullptr);
+    assert(pIdentityOut != nullptr && *pIdentityOut == nullptr);
+
+    if (items == nullptr)
+    {
+        return kErrOutItemsNull;
+    }
+
+    CFIndex itemCount = CFArrayGetCount(items);
+
+    if (itemCount == 0)
+    {
+        return kErrOutItemsEmpty;
+    }
+
+    CFTypeRef bestItem = nullptr;
+
+    printf("Dictionary Type: %lu\n", CFDictionaryGetTypeID());
+    printf("Certificate Type: %lu\n", SecCertificateGetTypeID());
+    printf("Identity Type: %lu\n", SecIdentityGetTypeID());
+
+    for (CFIndex i = 0; i < itemCount; i++)
+    {
+        CFTypeRef current = CFArrayGetValueAtIndex(items, i);
+        auto currentItemType = CFGetTypeID(current);
+
+        printf("Item %lu/%lu: %lu\n", i, itemCount, currentItemType);
+
+        if (currentItemType == SecIdentityGetTypeID())
+        {
+            bestItem = current;
+            break;
+        }
+        else if (bestItem == nullptr && currentItemType == SecCertificateGetTypeID())
+        {
+            bestItem = current;
+        }
+    }
+
+    if (bestItem == nullptr)
+    {
+        return -13;
+    }
+
+    if (CFGetTypeID(bestItem) == SecCertificateGetTypeID())
+    {
+        CFRetain(bestItem);
+        *pCertOut = reinterpret_cast<SecCertificateRef>(const_cast<void*>(bestItem));
+        return 1;
+    }
+
+    if (CFGetTypeID(bestItem) == SecIdentityGetTypeID())
+    {
+        CFRetain(bestItem);
+        *pIdentityOut = reinterpret_cast<SecIdentityRef>(const_cast<void*>(bestItem));
+	//if (*pIdentityOut == nullptr)
```

Comment:
This was me trying to see if I could remove the call.  It didn't work.  It's on my pre-merge cleanup list.